src/main/java/ee/tuleva/onboarding/payment/provider/montonio/MontonioOrderCreator.java

```diff
@@ -0,0 +1,127 @@
+package ee.tuleva.onboarding.payment.provider.montonio;
+
+import ee.tuleva.onboarding.auth.principal.Person;
+import ee.tuleva.onboarding.currency.Currency;
+import ee.tuleva.onboarding.locale.LocaleService;
+import ee.tuleva.onboarding.payment.PaymentData;
+import ee.tuleva.onboarding.payment.PaymentData.PaymentType;
+import ee.tuleva.onboarding.payment.provider.PaymentInternalReferenceService;
+import java.math.BigDecimal;
+import java.time.Clock;
+import java.util.Locale;
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class MontonioOrderCreator {
+
+  private final Clock clock;
+
+  private final PaymentInternalReferenceService paymentInternalReferenceService;
+
+  private final MontonioPaymentChannelConfiguration montonioPaymentChannelConfiguration;
+
+  private final LocaleService localeService;
+
+  @Value("${payment-provider.use-fake-notification-url}")
+  private Boolean useFakeNotificationsUrl;
+
+  @Value("${api.url}")
+  private String apiUrl;
+
+  @Value("${payment.member-fee}")
+  private BigDecimal memberFee;
+
+  @Value("${payment.member-fee-test-personal-code}")
+  private String memberFeeTestPersonalCode;
+
+  private static String getEpisPaymentDescription(PaymentData paymentData) {
+    if (paymentData.getType() == PaymentType.MEMBER_FEE) {
+      return String.format("member:%s", paymentData.getRecipientPersonalCode());
```

Comment:
Any way to use polymorphism here instead of if-else statements?

Reply:
Discussed during call, created ticket, will be done during follow-up